SCons/Job.py

```diff
@@ -76,19 +78,29 @@ def __init__(self, num, taskmaster):
         allocated.  If more than one job is requested but the Parallel
         class can't do it, it gets reset to 1.  Wrapping interfaces that
         care should check the value of 'num_jobs' after initialization.
+
+        'remote_cache' can be set to a RemoteCache.RemoteCache object.
+
+        'use_scheduler_v2' can be set to True to opt into the newer and more
+        aggressive scheduler.
         """
 
         self.job = None
-        if num > 1:
-            stack_size = explicit_stack_size
-            if stack_size is None:
-                stack_size = default_stack_size
 
-            try:
+        stack_size = explicit_stack_size
+        if stack_size is None:
+            stack_size = default_stack_size
+
+        try:
+            if ((remote_cache and remote_cache.fetch_enabled) or
+                    use_scheduler_v2):
+                self.job = ParallelV2(taskmaster, num, stack_size, remote_cache)
```

Comment:
could this be isolated into a tool and duck typed in when the tool is loaded? Then scons would not be forced to take on urllib3 dependency.

Something similar was done with the ninja tool: https://github.com/SCons/scons/blob/master/SCons/Tool/ninja/__init__.py#L382

its a bit hacky but these internal class don't really have any other exposure.

Reply:
FWIW urllib3 isn't strictly required by SCons with my change; it is still only required if you enable remote caching. Otherwise, you can still run SCons without urllib3.

I am honestly not sure about the idea of putting the new scheduler in a tool. It may also depend on whether people want this ParallelV2 to replace Parallel as a new-and-improved scheduler.